vpr/src/route/bucket.h

```diff
@@ -257,18 +259,44 @@ class Bucket : public HeapInterface {
 
     size_t seed_; /* Seed for fast_rand, should be non-zero */
 
-    BucketItem** heap_; /* Buckets for linked lists*/
-    size_t heap_size_;  /* Number of buckets */
-    size_t heap_head_;  /* First non-empty bucket */
-    size_t heap_tail_;  /* Last non-empty bucket */
-    float conv_factor_; /* Cost bucket scaling factor */
+    BucketItem** heap_;      /* Buckets for linked lists*/
+    size_t heap_size_;       /* Number of buckets */
+    size_t heap_head_;       /* First non-empty bucket */
+    size_t heap_tail_;       /* Last non-empty bucket */
+    float conv_factor_;      /* Cost bucket scaling factor */
+    float division_scaling_; /* Scaling factor used during rescaling.
+                              * Larger division scaling results in larger
+                              * conversion factor.
+                              */
+    ssize_t max_buckets_;    /* Maximum number of buckets to control when to
```

Comment:
Explaining how division_scaling and max_buckets interact would be useful.  I can imagine setting a max_buckets and a max_cost and computing conv_factor.  This is a bit more involved than that; can you give an example of how the parameters interact?